src/Components/Server/src/Builder/RazorComponentsEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Server;
+using Microsoft.AspNetCore.Routing;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    /// <summary>
+    /// Extensions for <see cref="RazorComponentsEndpointBuilder"/>
+    /// </summary>
+    public static class RazorComponentsEndpointRouteBuilderExtensions
+    {
+        /// <summary>
+        /// Maps the SignalR Components Hub to the path <see cref="ComponentsHub.DefaultPath"/> and associates
+        /// the component <typeparamref name="TComponent"/> to this hub instance.
+        /// The DOM selector for <typeparamref name="TComponent"/> is the type name in lowercase.
+        /// </summary>
+        /// <typeparam name="TComponent">The first <see cref="IComponent"/> associated with this <see cref="ComponentsHub"/>.</typeparam>
+        /// <param name="routes">The <see cref="RouteBuilder"/>.</param>
+        /// <returns>The <see cref="RazorComponentsEndpointBuilder"/>.</returns>
+        public static RazorComponentsEndpointBuilder MapComponentsHub<TComponent>(
```

Comment:
Change the name to `MapRootComponent` hubs are an implementation detail.

Reply:
this comment courtesy of @davidfowl  - my suggestion is `MapComponent`. We both don't like the inclusion of `Hub` in the name.

Reply:
Both things are weird. 
* MapComponent is weird because there can be more than one chained, so `routes.MapComponent<App>().AddComponent<Second>()` is weird.
* MapRootComponent is also weird because the word Root has a specific meaning IMO in the components/html world where it implies hierarchy, so its weird to have MapRootComponent and then AddComponent after that (when they are in reality siblings)

I don't have a suggestion right now that I like, but it's 8:30 in the morning, so something will come up during the day.

Reply:
`.MapComponents().AddComponent<First>().AddComponent<Second>()` ... if `MapComponents` is just a way of getting the required type of builder.

Or just `routes.AddComponent<First>().AddComponent<Second>()`, if on each call we can check if it's the first, and then conditionally wire up the one-time stuff only in the first call.